src/app/services/canvas.service.ts

```diff
@@ -0,0 +1,276 @@
+            );
+            break
+          // healing animations
+          case AnimationType.heal:
+            target.x(
+              100 * Math.sin((frame.time * 2 * Math.PI) / duration) + posX
+            );
+            target.y(
+              100 * Math.cos((frame.time* 2 * Math.PI) / duration) + possY
+            );
+            break
+        }
+      }
+    }, this.layer);
+    animation.start();
+    return animation;
+  }
+
+  initStartBtn(playService: PlayService, startText?: string) {
+    this.startBtnGroup.destroy();
+    this.startBtnGroup = new Konva.Group({
+      x: 490,
+      y: 300,
+    });
+    this.startBtnGroup.add(this.createRecLayer(0, 0));
+    this.startBtnGroup.add(this.createTextLayer(startText ? 16 : 35,20, startText || 'Start'));
+
+    this.startBtnGroup.on('click', () => {
+      playService.startBattle();
+      this.startBtnGroup.destroy();
+    });
+    this.layer.add(this.startBtnGroup);
+  }
+
+  initHero(hero: Hero, playService: PlayService) {
+    this.heroGroup.destroy();
+    this.heroGroup = new Konva.Group({
+      x: 100,
+      y: 350,
+    });
+    this.heroGroup.add(this.createImageLayer(0, 0, 250, 250, hero.avatar || hero.name));
+    this.heroGroup.on('click', () => {
+      if (playService.paused) {
+        playService.clearHero();
+        this.heroGroup.destroy();
+        this.heroInfoGroup.destroy();
+      }
+    });
+    this.initHeroInfo(hero);
+    this.layer.add(this.heroGroup);
+  }
+  // init HP, name
+  initHeroInfo(hero: Hero) {
+    this.heroInfoGroup.destroy();
+    this.heroInfoGroup = new Konva.Group({
+      x: 95,
+      y: 350,
+    });
+    let hpWidth = 160;
+    let hpColor = 'lightGreen';
+    let stroke = undefined;
+    let heroHPText = `Hp: ${Math.max(hero.health, 0)}/${hero.maxHp}`;
+    if (hero.maxHp && hero.health > 0) {
+      const ratio = hero.health/hero.maxHp;
+      hpWidth = 160 * ratio;
+      if (ratio < 0.5) { //show red when <50%
+        hpColor = 'red';
+        stroke = 'red';
+      }
+    }
+    if (hero.health<=0) {
+      hpColor = 'red';
+      stroke = 'red';
+      heroHPText = `Hero Fainted`;
+    }
+    this.heroInfoGroup.add(this.createRecLayer(30,250, {width: 200, height: 70, fill: '#F6F5F5', opacity: 0.2, stroke}));
+    this.heroInfoGroup.add(this.createTextLayer(80,260, `${hero.name} - Lv.${hero.level}`, {fill: '#0C359E'}));
+    this.heroInfoGroup.add(this.createTextLayer(90,280, heroHPText, {fill: '#0C359E'}));
+    this.heroInfoGroup.add(this.createRecLayer(50,300, {width: hpWidth, height: 10, fill: hpColor}));
+    this.layer.add(this.heroInfoGroup);
+  }
+
+  initMonster(monster: Monster) {
+    this.monsterGroup.destroy();
+    this.monsterGroup = new Konva.Group({
+      x: 700,
+      y: 200,
+    });
+    this.monsterGroup.add(this.createImageLayer(0, 0, 250, 250, monster.avatar || monster.name));
+    this.initMonsterInfo(monster);
+    this.layer.add(this.monsterGroup);
+  }
```

Comment:
Better abstraction to add these as their own classes (but it is listed in the README so you already caught this which makes me glad)